src/test/ui/error-codes/E0004-2.stderr

```diff
@@ -1,8 +1,8 @@
-error[E0004]: non-exhaustive patterns: multiple patterns of type `std::option::Option<i32>` are not handled
+error[E0004]: non-exhaustive patterns: `None` and `Some(_)` not covered
   --> $DIR/E0004-2.rs:4:11
    |
 LL |     match x { }
-   |           ^
+   |           ^ patterns `None` and `Some(_)` not covered
    |
    = help: ensure that all possible cases are being handled, possibly by adding wildcards or more match arms
```

Comment:
Some type information is lost here, so perhaps we could add something like:
```suggestion
   = help: ensure that all possible cases are being handled, possibly by adding wildcards or more match arms
   = note: the matched value is of type `std::option::Option<i32>`
```

Reply:
Other cases of non-exhaustive matches don't give info about the type either. I feel that since we suggest missing constructors, users have enough info and don't really need that extra type info.

Reply:
My suggestion is to do it for *all* non-exhaustive matches (but not necessarily in this PR). The user might not be so familiar with the code base to understand what the type of a constructor is.

Reply:
I'll leave this for a possible future PR then

Reply:
Can you file a ticket for this follow up work?

Reply:
Filed https://github.com/rust-lang/rust/issues/67259.